### ISSUE TITLE
Develop Ch. 05 Gradient descent and backpropagation

### ISSUE THREAD

**Issue by louishhy** [louishhy/30min-of-deep-learning#8]
## 📘 Target Section

05 - Gradient descent and backprop

## 🛠️ Planned Content

Introduce the backprop mechanism in PyTorch.
Will not cover much on formula but more focused on how PyTorch implement this and how to use APIs.

Include:
- PyTorch Tensors and gradients (how are they implemented?)
- Common pitfalls regarding torch backprops (`eval`, `no_grad` etc.)

More suggestions are welcomed.
